paasta_tools/kubernetes_tools.py

```diff
@@ -368,32 +388,72 @@ def get_kubernetes_containers(
                     container_port=8888,
                 ),
             ],
-            volume_mounts=self.get_volume_mounts(volumes=volumes),
+            volume_mounts=self.get_volume_mounts(
+                docker_volumes=docker_volumes,
+                aws_ebs_volumes=aws_ebs_volumes,
+            ),
         )
         containers = [service_container] + self.get_sidecar_containers(system_paasta_config=system_paasta_config)
         return containers
 
-    def get_pod_volumes(self, volumes: Sequence[DockerVolume]) -> Sequence[V1Volume]:
+    def get_pod_volumes(
+        self,
+        docker_volumes: Sequence[DockerVolume],
+        aws_ebs_volumes: Sequence[AwsEbsVolume],
+    ) -> Sequence[V1Volume]:
         pod_volumes = []
-        for volume in volumes:
+        unique_docker_volumes = {
+            self.get_docker_volume_name(docker_volume): docker_volume
+            for docker_volume in docker_volumes
+        }
+        for name, docker_volume in unique_docker_volumes.items():
             pod_volumes.append(
                 V1Volume(
                     host_path=V1HostPathVolumeSource(
-                        path=volume['hostPath'],
+                        path=docker_volume['hostPath'],
                     ),
-                    name=self.get_sanitised_volume_name(volume['containerPath']),
+                    name=name,
+                ),
+            )
+        unique_aws_ebs_volumes = {
+            self.get_aws_ebs_volume_name(aws_ebs_volume): aws_ebs_volume
+            for aws_ebs_volume in aws_ebs_volumes
+        }
+        for name, aws_ebs_volume in unique_aws_ebs_volumes.items():
+            pod_volumes.append(
+                V1Volume(
+                    aws_elastic_block_store=V1AWSElasticBlockStoreVolumeSource(
+                        volume_id=aws_ebs_volume['volume_id'],
+                        fs_type=aws_ebs_volume.get('fs_type'),
+                        partition=aws_ebs_volume.get('partition'),
+                        # k8s wants RW volume even if it's later mounted RO
+                        read_only=False,
+                    ),
+                    name=name,
                 ),
             )
         return pod_volumes
 
-    def get_volume_mounts(self, volumes: Sequence[DockerVolume]) -> Sequence[V1VolumeMount]:
+    def get_volume_mounts(
+        self,
+        docker_volumes: Sequence[DockerVolume],
+        aws_ebs_volumes: Sequence[AwsEbsVolume],
+    ) -> Sequence[V1VolumeMount]:
         volume_mounts = []
-        for volume in volumes:
+        for docker_volume in docker_volumes:
```

Comment:
this function could be DRY'd a little - can you:

- rename DockerVolume.containerPath to DockerVolume.container_path
- move logic for how to rename ``DockerVolume`` vs ``AwsEbsVolume`` to  ``get_aws_ebs_volume_name``
- reduce this function to 'just' be:
```
[V1VolumeMount(
     mount_path=aws_ebs_volume['container_path'],
     name=self.get_aws_ebs_volume_name(aws_ebs_volume),
     read_only=self.read_only_mode(aws_ebs_volume),
) for volume in aws_ebs_volumes + docker_volumes]
```

Reply:
`containerPath` is actually a name of the property in our configs from `srv-configs`, we cannot rename that easily.

Reply:
Also `get_aws_ebs_volume_name()` can't be used for `DockerVolume` instances.